### ISSUE TITLE
oxford_flowers102 bad splits

### ISSUE THREAD

**Issue by TeaPearce** [tensorflow/datasets#3022]
The train/val/test splits in the tfds [oxford_flowers102](https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/image_classification/oxford_flowers102.py) don't match up with established splits.

Training on the train and val splits, only acheive around 91% accuracy with finetuning. Should acheive 98%, e.g. table 6 [here](https://arxiv.org/abs/1905.11946). If one reads in the entire dataset and creates a random split, this is acheivable. Has also been noted on stackoverflow [here](https://stackoverflow.com/questions/60646972/trouble-with-splitting-data-from-tensorflow-datasets).

**Comment by jatin-code777** [comment]
This is not an issue within TFDS itself but perhaps a bug in the dataset itself.
I redownloaded the original dataset and confirmed that the splits given in the dataset match those in TFDS.
![image](https://user-images.githubusercontent.com/33352338/107138782-68ac2b00-693c-11eb-9dee-4f8a6394dc02.png)


These also match with the table 6 in the [paper](https://arxiv.org/abs/1905.11946):
![image](https://user-images.githubusercontent.com/33352338/107138881-4b2b9100-693d-11eb-8672-85d4cd89dbce.png)

In my opinion, the fix for this should come from the dataset itself.

**Comment by vijayphoenix** [comment]
A workaround would be to do something like this

```python

>>> import tensorflow_datasets as tfds
>>> test, train, validation = tfds.load('oxford_flowers102', split=['train', 'test', 'validation'])

>>> sum(1 for _ in train)
6149
```

Perhaps, we can add a docstring in the TFDS to warn the users

**Comment by Conchylicultor** [comment]
TFDS provide the datasets as close as the original datasets authors. As pointed out above, TFDS splits match the splits as defined by the Oxford author. So I'm making this bug as working as intended.

Note: Our documentation already provide the number of examples: https://www.tensorflow.org/datasets/catalog/oxford_flowers102

Or programatically:
```python
info = tfds.builder('oxford_flowers102)
info.split['test'].num_examples
```
Or
```python
test, train, validation = tfds.load('oxford_flowers102', split=['train', 'test', 'validation'])
print(len(train))
```

**Comment by TeaPearce** [comment]
Thank all, sounds sensible. I don't know the history of when/how/why the dataset splits evolved, but wanted to document it somewhere.

**Comment by vijayphoenix** [comment]
You can find the splitting and slicing doc here
https://www.tensorflow.org/datasets/splits
